packages/keymaps/src/browser/keybindings-widget.tsx

```diff
@@ -0,0 +1,202 @@
+/********************************************************************************
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import React = require('react');
+import { injectable, inject, postConstruct } from 'inversify';
+import { CommandRegistry, Command, MessageService } from '@theia/core/lib/common';
+import { ReactWidget } from '@theia/core/lib/browser/widgets/react-widget';
+import { KeybindingRegistry } from '@theia/core/lib/browser';
+
+export interface KeybindingItem {
+    id: string,
+    command: string,
+    keybinding?: string,
+    context?: string,
+}
+
+@injectable()
+export class KeybindingWidget extends ReactWidget {
+
+    @inject(CommandRegistry)
+    protected readonly commandRegistry: CommandRegistry;
+
+    @inject(KeybindingRegistry)
+    protected readonly keybindingRegistry: KeybindingRegistry;
+
+    @inject(MessageService)
+    protected readonly messageService: MessageService;
+
+    protected items: KeybindingItem[];
+
+    constructor() {
+        super();
+        this.id = 'keybindings.view.widget';
+        this.title.label = 'View Keybindings';
+        this.title.closable = true;
+        this.update();
+    }
+
+    @postConstruct()
+    protected init(): void {
+        this.items = this.getItems();
+    }
+
+    protected searchKeybindings(): void {
+        this.items = this.getItems();
+        const searchField = this.findSearchField();
+        const query = searchField ? searchField.value.trim().toLocaleLowerCase() : '';
+        this.items = this.items.filter(a => this.searchFilter(a, query));
+        this.update();
+    }
+
+    protected findSearchField(): HTMLInputElement | null {
+        return document.getElementById('search-keybindings') as HTMLInputElement;
+    }
+
+    protected searchFilter(item: KeybindingItem, query: string): boolean {
```

Comment:
@akosyakov I'm using the fuzzy package as of my latest changes but was having a lot of difficulty processing the data afterwards (regexp, building html from my data) in order to adjust the formatting of the table to display results in `bold` and `blue`. I'm wondering if there is a simple way to do it without any performance degradation.

![selection_023](https://user-images.githubusercontent.com/40359487/45840382-d2ccc280-bce4-11e8-9370-da416e25126c.png)

Reply:
Let's leave it out for a separate PR? If you want to me look how it can be optimized please push your changes.

Reply:
I can include fuzzy searching but perhaps keep the highlighting as part of another PR if it can't be optimised. I pushed the changes, but please note the code might be really ugly as of now :)